Packs/CiscoAMP/Integrations/CiscoAMP/CiscoAMP.py

```diff
@@ -0,0 +1,3538 @@
+            'group_guid[]': group_guids,
+            'start_time': start_time,
+            'end_time': end_time,
+            'limit': limit,
+            'offset': offset,
+        })
+
+        return self._http_request(
+            method='GET',
+            url_suffix=f'/vulnerabilities/{sha256}/computers',
+            params=params,
+        )
+
+
+''' COMMAND FUNCTIONS '''  # pylint: disable=pointless-string-statement
+
+
+def fetch_incidents(
+    client: Client,
+    last_run: Dict[str, Any],
+    first_fetch_time: str,
+    event_types: List[int] = None,
+    incident_severities: List[str] = None,
+    max_incidents_to_fetch: int = FETCH_LIMIT,
+) -> Tuple[Dict[str, int], List[dict]]:
+    """
+    Retrieves new alerts every interval (default is 1 minute).
+    Implements the logic of making sure that incidents are fetched only once.
+    By default it's invoked by XSOAR every minute.
+    It will use last_run to save the time of the last incident it processed and previous incident IDs.
+    If last_run is not provided, first_fetch_time will be used to determine when to start fetching the first time.
+
+    Args:
+        client (Client): Cisco AMP client to run desired requests
+        last_run (Dict[str, Any]):
+            last_fetch: Time of the last processed incident.
+            previous_ids: List of incident IDs to that would not be repeated.
+        first_fetch_time (str): Determines the time of when fetching has been started.
+        event_types (List[int], optional): Event types to filter by.
+            Defaults to None.
+        incident_severities (List[str], optional): Incident severities to filter by.
+            Defaults to None.
+        max_incidents_to_fetch (int, optional): Max number of incidents to fetch in a single run.
+            Defaults to FETCH_LIMIT.
+
+    Returns:
+        Tuple[Dict[str, int], List[dict]]:
+            next_run: Contains information that will be used in the next run.
+            incidents: List of incidents that will be created in XSOAR.
+    """
+    last_fetch = last_run.get('last_fetch')
+    previous_ids = last_run.get('previous_ids', [])
+
+    # If a last fetch run doesn't exist, use the first fetch time.
+    if last_fetch is None:
+        last_fetch = first_fetch_time
+
+    last_fetch_timestamp = date_to_timestamp(last_fetch, ISO_8601_FORMAT)
+
+    response = client.event_list_request(
+        start_date=last_fetch,
+        event_types=event_types
+    )
+    items = response['data']
+
+    incidents: List[Dict[str, Any]] = []
+    incident_name = 'Cisco AMP Event ID:"{event_id}"'
+
+    for item in items:
+        # Break once the maximum number of incidents has been achieved.
+        if len(incidents) >= max_incidents_to_fetch:
+            break
+
+        # Continue if the incident severity isn't in the requested list (only if there is one).
+        if incident_severities and (severity := item.get('severity')) not in incident_severities:
```

Comment:
Nope, not always.
I've only made them be fetched if the user hasn't entered any severity.
Should I always fetch items without severities even if the user has entered a specific severity?

Reply:
I'm not sure if this is needed.

In the first and third  `bool` the empty `list` and `None` will already result in `False`, thus the statement won't continue. This will prevent from getting `True` in the first and an exception in the third (`None` is not `iterable`).

In the second `bool` we get `True` because the `list` isn't empty and `None` isn't in that `list`.

```
>>> bool([] and None not in [])
False
>>> bool(['Low'] and None not in ['Low'])
True
>>> bool(None and None not in None)
False
```